central/reprocessor/reprocessor.go

```diff
@@ -460,9 +462,28 @@ func (l *loopImpl) reprocessNodes() {
 }
 
 func (l *loopImpl) reprocessWatchedImage(name string) bool {
-	img, err := imageEnricher.EnrichImageByName(emptyCtx, l.imageEnricher, imageEnricher.EnrichmentContext{
+	enrichmentCtx := imageEnricher.EnrichmentContext{
 		FetchOpt: imageEnricher.IgnoreExistingImages,
-	}, name)
+	}
+
+	ctx := emptyCtx
+	if features.DelegateWatchedImageReprocessing.Enabled() {
+		// Delegated scanning performs an access check, an empty context will
+		// trigger a hard panic. If an image watch was successfully created
+		// we can assume at that time the user had access, so we create a
+		// context that has access to delegate the scan.
+		ctx = sac.WithGlobalAccessScopeChecker(
+			ctx,
+			sac.AllowFixedScopes(
+				sac.AccessModeScopeKeys(storage.Access_READ_ACCESS),
+				sac.ResourceScopeKeys(resources.Image),
+			),
+		)
+
+		enrichmentCtx.Delegable = true
```

Comment:
just confirming this is saying that we allow it to be delegated, but it's not necessarily delegated, right?

Reply:
correct